### PR TITLE
Pub/Sub: Expose publish retry settings

### LINKED ISSUE #6641: Pub/Sub: publisher client's publish method does not expose retry settings
https://github.com/googleapis/google-cloud-python/blob/a5af6acb85d8e1ed97e56ad7fc1cb0aefc524e61/pubsub/google/cloud/pubsub_v1/publisher/client.py#L166

Publish retry is supported in [Node.js](https://cloud.google.com/nodejs/docs/reference/pubsub/0.20.x/v1.PublisherClient#publish) and [Java](https://github.com/googleapis/google-cloud-java/blob/eb9fae59a236773a38c87dbce82206d4a267bf76/google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java#L107). It should be supported by Python too.

### PR DESCRIPTION
**Pull request by anguillanneuf**
This closes #6641. 

This PR updates the GAPIC publisher client using `synth.py`.

### CODE REVIEW

File: pubsub/synth.py [hunk 1]
```diff
@@ -109,6 +109,63 @@
...
+        else:
+            d1[k] = v2
+    return d1
+    \n\n\g<0>"""
+)
+
+s.replace(
+    "google/cloud/pubsub_v1/gapic/publisher_client.py",
+    "client_config \(dict\): DEPRECATED.",
+    "client_config \(dict\):"
```
Comment (busunkim96) [inline pubsub/synth.py]: You can remove the `\` on this line. The replacement string is treated as a regular string, not a regex.

Comment (anguillanneuf) [reply pubsub/synth.py]: Thanks for catching this one!

File: pubsub/noxfile.py [hunk 2]
```diff
@@ -23,12 +23,6 @@
 
 LOCAL_DEPS = (os.path.join("..", "api_core"), os.path.join("..", "core"))
 
-BLACK_PATHS = ["docs", "google", "tests", "noxfile.py", "setup.py"]
```
Comment (busunkim96) [inline pubsub/noxfile.py]: Hey @anguillanneuf I see recent changes to our templates being undone here. Would you mind installing synthtool from head and re-running? 

```python3 -m pip install --user --upgrade git+https://github.com/googleapis/synthtool.git```

Thank you!

Comment (anguillanneuf) [reply pubsub/noxfile.py]: I just used the latest synthtool. How does it look now?

File: pubsub/noxfile.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def default(session):
         "--cov-append",
         "--cov-config=.coveragerc",
         "--cov-report=",
-        "--cov-fail-under=97",
+        "--cov-fail-under=0",
```
Comment (plamut) [inline pubsub/noxfile.py]: Sanity check - we are proceeding with this? I saw [the issue](https://github.com/googleapis/google-cloud-python/issues/8174) mentioning the problem with any threshold below 100 %, but just want to make sure that this is indeed an intentional change.

Comment (busunkim96) [reply pubsub/noxfile.py]: :+1: We merged the template change yesterday to synthtool.

**Comment by plamut** [comment]
Based on @busunkim96's comment, this seems to be good for merging. :+1:

**Pull request merged**